doc/us-en/tutorials/Device-connect-to-OpenEdge-with-hub-module.md

```diff
@@ -1,3 +1,149 @@
+
+**Tip**：The hub.baidubce.com/openedgedev/openedge-agent:latest image in the log is the Agent Module published by Openedge Cloud Management Suite.
+
+### OpenEdge Connection Test
+
+The configuration of OpenEdge Master is as follows:
+
+```yaml
+version: V2
+services:
+  - name: localhub
+    image: 'hub.baidubce.com/openedgedev/openedge-hub:latest'
+    replica: 1
+    ports:
+      - '1883:1883'
+      - '8080:8080'
+      - '8883:8883'
+    env: {}
+    mounts:
+      - name: dxc_localhub_conf-V2
+        path: etc/openedge
+        readonly: true
+      - name: dxc_localhub_cert-V1
+        path: var/db/openedge/cert
+        readonly: true
+      - name: dxc_localhub_data-V1
+        path: var/db/openedge/data
+        readonly: true
+      - name: dxc_localhub_log-V1
+        path: var/log/openedge
+        readonly: true
+volumes:  
+  - name: dxc_localhub_conf-V2
+    path: var/db/openedge/dxc_localhub_conf/V2
+  - name: dxc_localhub_cert-V1
+    path: var/db/openedge/dxc_localhub_cert/V1
+  - name: dxc_localhub_data-V1
+    path: var/db/openedge/dxc_localhub_data
+  - name: dxc_localhub_log-V1
+    path: var/db/openedge/dxc_localhub_log  
+```
+
+The configuration of OpenEdge Hub Module is as follows:
+
+```yaml
+listen:
+  - tcp://0.0.0.0:1883
+  - ssl://0.0.0.0:8883
+  - ws://0.0.0.0:8080/mqtt
+certificate:
+  ca: var/db/openedge/cert/ca.pem
+  cert: var/db/openedge/cert/server.pem
+  key: var/db/openedge/cert/server.key
+principals:
+  - username: two-way-tls
+    permissions:
+      - action: 'pub'
+        permit: ['tls/#']
+      - action: 'sub'
+        permit: ['tls/#']
+  - username: test
+    password: hahaha
+    permissions:
+      - action: 'pub'
+        permit: ['#']
+      - action: 'sub'
+        permit: ['#']
+logger:
+  path: var/log/openedge/service.log
+  level: "debug"
+```
+
+As mentioned above, when the Hub Module starts, it will open ports 1883, 8883 and 8080 at the same time, which used them as TCP, SSL, WS (Websocket) connection ports. And we will use MQTTBOX and MQTT.fx as MQTT Client to connect to OpenEdge . 
+
+**TCP Connection Test**
+
+Start the MQTT.fx, enter the Edit Connection Profiles page, fill in the Profile Name, fill in the Broker Address and Port according to the connection configuration of the OpenEdge Hub module, and then configure the username & password in User Credentials according to the `principals` configuration option. Then click `Apply` button to complete the connection configuration of MQTT.fx with TCP protocal as shown below.
+
+![TCP connection configuration](../../images/tutorials/connect/mqttbox-tcp-connect-config.png)
+
+Then close the configuration page, select the Profile Name configured, then click `Connect` button, if the connection configuration information matches the permission information allowed in the `principals` configuration item of OpenEdge Hub module, you can see the connection success flag, as shown below .
```

Comment:
Then close the configuration page, select the Profile Name configured, then click `Connect` button, if the connection configuration information matches the `principals` configuration of OpenEdge Hub module, you can see the connection success flag which as shown below.